metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/DepublishRecordIdService.java

```diff
@@ -178,4 +189,24 @@ public int addRecordIdsToBeDepublished(MetisUser metisUser, String datasetId,
     result.setResultsAndLastPage(records, depublishRecordIdDao.getPageSize(), page);
     return result;
   }
+
+  public WorkflowExecution createDepublishWorkflowExecution(MetisUser metisUser, String datasetId,
+      boolean datasetDepublish, int priority) throws GenericMetisException {
+    // Authorize.
+    authorizer.authorizeReadExistingDatasetById(metisUser, datasetId);
+
+    //Prepare depublish workflow, do not save in the database. Only create workflow execution
+    final Workflow workflow = new Workflow();
+    workflow.setDatasetId(datasetId);
+    final DepublishPluginMetadata depublishPluginMetadata = new DepublishPluginMetadata();
+    depublishPluginMetadata.setEnabled(true);
+    depublishPluginMetadata.setDatasetDepublish(datasetDepublish);
+    workflow.setMetisPluginsMetadata(Collections.singletonList(depublishPluginMetadata));
+
+    // Validate the new workflow.
+    workflowUtils.validateWorkflowPlugins(workflow, null);
```

Comment:
This is not necessary I think. This already happens when calling `orchestratorService.addWorkflowInQueueOfWorkflowExecutions`?

Reply:
Indeed! Removed it